### ISSUE TITLE
docker-compose.yml

### ISSUE THREAD

**Issue by Ninja-FSE** [ahembree/ansible-hms-docker#7]
Think i found a small bug in the docker-compose.yml.j2

If i just enable authentik, portainer and traefik Tautulli's ports are still beeing added to docker-compose.yml
here is a snippet from only Traefik, Authentik and portainer enabled.

Problem is that if i enable sonarr or any other app, that app is getting this ports, 8181.

```
version: '3'
services:
  # Portainer container, webgui for docker
  portainer:
    image: portainer/portainer-ce:latest
    container_name: portainer
    command: -H unix:///var/run/docker.sock
    restart: unless-stopped
    logging:
      driver: json-file
      options:
        max-size: 10m
    networks:
      - "traefik_net"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/REDACT/apps/portainer/config:/data
    labels:
      - traefik.enable=true
      - traefik.http.services.portainer-REDACT.loadbalancer.server.port=9000
      - traefik.http.routers.portainer-REDACT.rule=Host(`portainer.REDACT`)
      - traefik.http.routers.portainer-REDACT.middlewares=internal-ipwhitelist

  # Watchtower container, automatic updates
  watchtower:
    image: containrrr/watchtower:latest
    container_name: watchtower
    restart: unless-stopped
    logging:
      driver: json-file
      options:
        max-size: 10m
    networks:
      - "traefik_net"
    command: --cleanup --schedule "0 0 9 * * *"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  # Traefik container, loadbalancer/reverse proxy/ssl
  traefik:
    image: traefik:v2.6.1
    container_name: traefik
    restart: unless-stopped
    logging:
      driver: json-file
      options:
        max-size: 10m
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    environment:
      - TZ=Europe/Stockholm
      - PUID=100
      - PGID=99
      - CF_DNS_API_TOKEN=REDACT
      - CF_ZONE_API_TOKEN=REDACT
    networks:
      - "traefik_net"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/REDACT/apps/traefik/config/traefik.yml:/etc/traefik/traefik.yml
      - /opt/REDACT/apps/traefik/config/certs/:/certs/
    labels:
      - traefik.enable=true
      - traefik.http.routers.traefik-REDACT.rule=Host(`traefik.REDACT`)
      - traefik.http.services.traefik-REDACT.loadbalancer.server.port=8080
      - "traefik.http.middlewares.internal-ipwhitelist.ipwhitelist.sourcerange=127.0.0.1/32, 192.168.56.0/24, 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16"
      - "traefik.http.middlewares.external-ipwhitelist.ipwhitelist.sourcerange=0.0.0.0/0"
      - "traefik.http.routers.traefik-REDACT.middlewares=internal-ipwhitelist"

  # Authentik container, authentication/authorization
  authentik-postgresql:
    container_name: authentik-postgresql
    image: postgres:12-alpine
    restart: unless-stopped
    networks:
      - authentik_net
    logging:
      driver: json-file
      options:
        max-size: 10m
    healthcheck:
      test: ["CMD", "pg_isready"]
      start_period: 20s
      interval: 30s
      retries: 5
      timeout: 5s
    volumes:
      - authentik_database:/var/lib/postgresql/data
    env_file:
      - .env
    environment:
      - POSTGRES_PASSWORD=${PG_PASS:?database password required}
      - POSTGRES_USER=${PG_USER:-REDACT}
      - POSTGRES_DB=${PG_DB:-REDACT}

  authentik-redis:
    container_name: authentik-redis
    image: redis:alpine
    restart: unless-stopped
    networks:
      - authentik_net
    logging:
      driver: json-file
      options:
        max-size: 10m
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping | grep PONG"]
      start_period: 20s
      interval: 30s
      retries: 5
      timeout: 3s

  authentik-server:
    container_name: authentik-server
    image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2022.5.2}
    restart: unless-stopped
    networks:
      - authentik_net
      - traefik_net
    logging:
      driver: json-file
      options:
        max-size: 10m
    command: server
    env_file:
      - .env
    environment:
      AUTHENTIK_REDIS__HOST: REDACT
      AUTHENTIK_POSTGRESQL__HOST: REDACT
      AUTHENTIK_POSTGRESQL__USER: ${PG_USER:-REDACT}
      AUTHENTIK_POSTGRESQL__NAME: ${PG_DB:-REDACT}
      AUTHENTIK_POSTGRESQL__PASSWORD: ${PG_PASS}
      # AUTHENTIK_ERROR_REPORTING__ENABLED: "true"
      # WORKERS: 2
    volumes:
      - /opt/REDACT/apps/authentik/media:/media
      - /opt/REDACT/apps/authentik/custom-templates:/templates
      - authentik_geoip:/geoip
    labels:
      - traefik.enable=true
      - traefik.http.services.authentik-server-REDACT.loadbalancer.server.port=9000
      - traefik.http.routers.authentik-server-REDCAT.rule=Host(`authentik.REDACT`)

  authentik-worker:
    container_name: authentik-worker
    image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2022.5.2}
    restart: unless-stopped
    networks:
      - authentik_net
    command: worker
    env_file:
      - .env
    environment:
      AUTHENTIK_REDIS__HOST: REDACT
      AUTHENTIK_POSTGRESQL__HOST: REDACT
      AUTHENTIK_POSTGRESQL__USER: ${PG_USER:-REDACT}
      AUTHENTIK_POSTGRESQL__NAME: ${PG_DB:-REDACT}
      AUTHENTIK_POSTGRESQL__PASSWORD: ${PG_PASS}
      # AUTHENTIK_ERROR_REPORTING__ENABLED: "true"
    # This is optional, and can be removed. If you remove this, the following will happen
    # - The permissions for the /media folders aren't fixed, so make sure they are 1000:1000
    # - The docker socket can't be accessed anymore
    user: root
    volumes:
      - /opt/REDACT/apps/authentik/media:/media
      - /opt/REDACT/apps/authentik/certs:/certs
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/REDACT/apps/authentik/custom-templates:/templates
      - authentik_geoip:/geoip

  authentik-geoipupdate:
    container_name: authentik-geoipupdate
    image: "maxmindinc/geoipupdate:latest"
    networks:
      - authentik_net
    volumes:
      - "authentik_geoip:/usr/share/GeoIP"

  authentik-geoipupdate:
    container_name: authentik-geoipupdate
    image: "maxmindinc/geoipupdate:latest"
    networks:
      - authentik_net
    volumes:
      - "authentik_geoip:/usr/share/GeoIP"
    env_file:
      - .env
    environment:
      GEOIPUPDATE_ACCOUNT_ID: "REDACT"
      GEOIPUPDATE_LICENSE_KEY: "REDACT"
      GEOIPUPDATE_EDITION_IDS: "GeoLite2-City"
      GEOIPUPDATE_FREQUENCY: "8"










    ports:
      - 8181:8181
    volumes:
      - /opt/REDACT/apps/tautulli/config:/config
      # Plex logs location
      - /opt/REDACT/apps/plex/config/Library/Application Support/Plex Media Server/Logs:/plex_logs:ro

  # Cloudflare DDNS container
  cloudflare-ddns:
    image: oznu/cloudflare-ddns:latest
    container_name: cloudflare-ddns
    restart: 'unless-stopped'
    logging:
      driver: json-file
      options:
        max-size: 10m
    environment:
      - API_KEY=REDACT
      - ZONE=REDACT
      - DELETE_ON_STOP=false
      - SUBDOMAIN=overseerr
      - PROXIED=true

networks:
  "download_net":
    driver: bridge
    attachable: false
  "media_net":
    driver: bridge
    attachable: false
  "traefik_net":
    driver: bridge
    attachable: true
  "authentik_net":
    driver: bridge
    attachable: false

volumes:
  authentik_database:
    driver: local
  authentik_geoip:
    driver: local


```
As you can see i dont have Tautulli enabled but it still adding Tautulli ports to the compose file. From docker-compose.yml.j2 file if there is any problems here i dont know.

```
{% if container_enabled_tautulli %}
  # Tautulli container, analytics
  tautulli:
    image: tautulli/tautulli:latest
    container_name: tautulli
    restart: {{ container_restart_policy }}
    logging:
      driver: json-file
      options:
        max-size: 10m
    networks:
      - "media_net"
      - "traefik_net"
      - "authentik_net"
    environment:
      - PUID={{ container_uid }}
      - PGID={{ container_gid }}
      - TZ={{ container_timezone }}
{% if traefik_enabled_tautulli %}
    labels:
      - traefik.enable=true
      - traefik.http.services.tautulli-{{ project_name }}.loadbalancer.server.port=8181
      - traefik.http.routers.tautulli-{{ project_name }}.rule=Host(`{{ hms_docker_container_map['tautulli']['proxy_host_rule'] }}.{{ hms_docker_domain }}`)
{% if not expose_public_enabled_tautulli %}
      - "traefik.http.routers.tautulli-{{ project_name }}.middlewares=internal-ipwhitelist"
{% endif %}
{% if authentik_enabled_tautulli %}
      - traefik.http.routers.tautulli-{{ project_name }}.middlewares=authentik-proxy-{{ project_name }}-tautulli-router@docker
{% endif %}
{% endif %}
{% endif %}
{% if expose_ports_enabled_tautulli %}
    ports:
      - 8181:8181
{% endif %}
    volumes:
      - {{ hms_docker_apps_path }}/tautulli/config:/config
      # Plex logs location
      - {{ hms_docker_apps_path }}/plex/config/Library/Application Support/Plex Media Server/Logs:/plex_logs:ro
{% endif %}
```

**Comment by ahembree** [comment]
Thanks for this report! Because of this reported issue, I was able to dig into why my templating was awful and figured out some extra parameters I can pass the template resource.

So now the template has correct indentation and is easier to read! Since it was so confusing to read before, yes there was an issue where I did not have if statements lined up correctly, causing the template to be build incorrectly.

These should now be resolved in https://github.com/ahembree/ansible-hms-docker/commit/49a910510e36d6323248c74f9f304712a88c36a9
